src/services/UserService.ts

```diff
@@ -145,6 +157,30 @@ export class UserService {
 		return user.toJSONPrivate();
 	}
 
+	public async forgotPassword(userEmail: string): Promise<APIPrivateUser> {
+		return getConnection().transaction(async entityManager => {
+			if (!userEmail) throw new APIError(HttpCode.Forbidden, ForgotPasswordError.UserNotFound);
+			const user = await entityManager.findOneOrFail(User, { email: userEmail }).catch(() => Promise.reject(new APIError(HttpCode.NotFound, ForgotPasswordError.UserNotFound)));
+			if (user.accountStatus === AccountStatus.Unverified) throw new APIError(HttpCode.Forbidden, ForgotPasswordError.AccountUnverified);
+			return user.toJSONPrivate();
+		});
+	}
+
+	public async resetPassword(userID: string, data: PasswordResetDataToCreate): Promise<APIPrivateUser> {
+		return getConnection().transaction(async entityManager => {
+			if (!userID) throw new APIError(HttpCode.NotFound, ResetPasswordError.UserNotFound);
+			const user = await entityManager.findOneOrFail(User, { id: userID }).catch(() => Promise.reject(new APIError(HttpCode.NotFound, ResetPasswordError.UserNotFound)));
+			if (!data.password) {
```

Comment:
This check could be moved to the top of the block to avoid unnecessarily fetching the user only to find that their request was invalid.